### ISSUE TITLE
JitBuilder structs can't have struct members

### ISSUE THREAD

**Issue by Leonardo2718** [eclipse/omr#390]
Currently, there is no way of defining the following structure using JitBuilder:

``` c
struct Outer {
   short outerField;
   struct {
      int innerField
   } i;
};
```

This is because the definition of `StructType::AddField()` (in `compiler/ilgen/TypeDictionary.cpp`) can only calculate the alignment of primitive types, as per the following snippet:

``` c++
TR::DataType primitiveType = typeInfo->getPrimitiveType();
uint32_t align = primitiveTypeAlignment[primitiveType] - 1;
_size = (_size + align) & (~align);
```

As a workaround, the struct element of the outer struct can be made a pointer to the inner struct.

``` c
struct Inner {
   int innerField
};

struct Outer {
   short outerField;
   Inner* i;
};
```

In cases where the inner struct is needed for alignment (e.g. when overlaying externally defined structs), the inner struct members must be "flattened" into the outer struct. This may require dummy fields to be added for correct alignment (see issue #376).

``` c
struct Outer {
   short outerField;
   short DUMMY_FIELD_FOR_ALIGNMENT;
   int i_innerField
};
```

**Comment by mstoodle** [comment]
You may want to adjust the phrasing "Currently, there is no way" ... "As a workaround" ...

**Comment by Leonardo2718** [comment]
This issue is resolved by the combination of #499 and #516.
